defaults/main.yml

```diff
@@ -1,2 +1,4 @@
 ---
 spark_version: "1.2.0+cdh5.3.*"
+
+spark_env_extras: []
```

Comment:
Potentially overkill, but I was experimenting with using plain maps for variables that are meant to represent key/value pairs (like environment variables) vs. arbitrarily structured maps:

```yaml
spark_env_extras:
  MESOS_NATIVE_LIBRARY: "/usr/local/lib/libmesos.so"
```

There are some downsides to nesting, but the upside is that this can be passed to any task via `environment`:

```yaml
- name: Something
  command: ./something.sh
  environment: spark_env_extras
```

`with_items` usage also changes to `with_dict`:

```yaml
- name: Configure application
  copy: content={{ item.value }}
        dest=/etc/spark/env/{{ item.key }}
  with_dict: spark_env_extras
```